### PR TITLE
Adding a better error message for user defined functions

### PR DESCRIPTION
**Pull request by ccoffrin**
An attempt to addresses #58.

### CODE REVIEW

**Comment by ccoffrin** [comment]
@odow @mlubin

**Comment by mlubin** [comment]
Is it easy to add a test case for this?

**Comment by ccoffrin** [comment]
Sure.  How does one catch warnings in Base.test?

**Comment by odow** [comment]
Once I merge the other PR and we switch to Base.Test instead of FactCheck: `@test_throws Exception func_that_throws()`

File: src/nl_write.jl [hunk 1]
```diff
@@ -255,4 +255,11 @@ end
 
 nl_variable(index::Integer) = "v$index"
 nl_number(value::Real) = "n$value"
-nl_operator(operator::Symbol) = "o$(func_to_nl[operator])"
+
+function nl_operator(operator::Symbol)
+    if !(operator in keys(func_to_nl))
```
Comment (odow) [inline src/nl_write.jl]: Not just `!haskey(func_to_nl, operator)`?

Comment (ccoffrin) [reply src/nl_write.jl]: Python habit where `key in dict` works.

**Pull request merged**